GUI/src/cz/fidentis/gui/actions/importfromimage/ImportFromImage.java

```diff
@@ -448,7 +456,36 @@ private Vector3f getUpdatedPosition(int vertexIdx, Model model) {
             return null;
         }
 
-        return gems;
+        // now from the depth maps selected by age and gender select at most N
+        // most similar faces to the input
+        // if there is not enoug GEMs selected just use all
+        if (numOfClosestGems >= gems.size()) {
+            return gems;
+        }
+        // otherwise limit the selected depth maps by procrustes distance:
+        Procrustes1ToMany procrustes = new Procrustes1ToMany(points, fps, true);
+        // register the considered depth maps to the input feature points
+        procrustes.align1withN();
+        // compute the procrustes distances of depth models to the input face
+        List<Double> distances = procrustes.compare1WithN(0.0f);
```

Comment:
tu by bolo asi lepsie ak by hodnota bola nastavena na 0.05f, aby to nahodou nezacyklilo.

Reply:
ok, zmenil som to na `0.05f`, ak nemas ine pripomienky mozes to mergnut